nginx/nginx.conf

```diff
@@ -8,8 +8,10 @@ http {
         location / {
             proxy_pass http://ckan:5000/;
         }
-        location /giftless {
+        location /giftless/ {
             proxy_pass http://giftless:6001/;
+            proxy_set_header X-Forwarded-Prefix /giftless;
+            proxy_set_header X-Forwarded-Host adr;
```

Comment:
As recommended by Datopian [here](https://github.com/datopian/giftless#fixing-generated-urls-when-running-behind-a-proxy), when running giftless behind a reverse proxy. 

Notice there is another coupling to the hostname `adr` in line 14 here. I wonder if there is anyway we can remove this?

Reply:
That's how I'm configuring reverse proxies, after discussing the issue with Michał some time ago:
```
    location / {
        add_header X-Proxy-Cache $upstream_cache_status;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Port $server_port;
        proxy_pass $upstream_domain;
    }
```
I have never used `X-Forwarded-Host` before. But I guess it should also be set to `$host` like `Host` in the example above.
For `X-Forwarded-Prefix` I thought you could use sth like `$location` to avoid duplication but I couldn't find if it's possible in the docs.

Also, consider adding a `server_name adr` in the configuration. I think that `adr` must come up somewhere, but it should be handled in the server_name part. It'd be great if Michał reviews the config afterwards.

Reply:
Ok I have updated these configs.  I will request @fulior review.